src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -86,7 +89,22 @@ public Client(String url, String username, String password) throws MalformedURLE
   public Client(URL url, String username, String password) throws MalformedURLException, URISyntaxException {
     this.rootUri = url.toURI();
 
-    this.rt = new RestTemplate(getRequestFactory(url, username, password));
+    this.rt = new RestTemplate(getRequestFactory(url, username, password, null, null));
+    this.rt.setMessageConverters(getMessageConverters());
+  }
+
+  /**
+   * Construct an instance with the provided url and credentials.
+   * @param url the url e.g. "http://localhost:15672/api/".
+   * @param username the user name.
+   * @param password the password
+   * @throws MalformedURLException for a badly formed URL.
+   * @throws URISyntaxException for a badly formed URL.
+   */
+  public Client(URL url, String username, String password, SSLConnectionSocketFactory sslConnectionSocketFactory, SSLContext sslContext) throws MalformedURLException, URISyntaxException {
```

Comment:
Should we instead have two constructors, one with an `SSLConnectionSocketFactory` and one with an `SSLContext`? I find that to be a more common way of expressing effectively mutually exclusive arguments.